internal/biz/resources/resources.go

```diff
@@ -16,55 +18,85 @@ import (
 )
 
 type ResourceRepository interface {
-	Save(context.Context, *model.Resource) (*model.Resource, error)
-	Update(context.Context, *model.Resource, uuid.UUID) (*model.Resource, error)
+	Create(context.Context, *model.Resource) (*model.Resource, []*model.Resource, error)
+	Update(context.Context, *model.Resource, uuid.UUID) (*model.Resource, []*model.Resource, error)
 	Delete(context.Context, uuid.UUID) (*model.Resource, error)
 	FindByID(context.Context, uuid.UUID) (*model.Resource, error)
 	FindByReporterResourceId(context.Context, model.ReporterResourceId) (*model.Resource, error)
+	FindByReporterData(context.Context, string, string) (*model.Resource, error)
 	ListAll(context.Context) ([]*model.Resource, error)
 }
 
+type InventoryResourceRepository interface {
+	FindByID(context.Context, uuid.UUID) (*model.InventoryResource, error)
+}
+
 var (
-	ErrResourceNotFound      = errors.New("resource not found")
-	ErrDatabaseError         = errors.New("db error while querying for resource")
-	ErrResourceAlreadyExists = errors.New("resource already exists")
+	ErrResourceNotFound             = errors.New("resource not found")
+	ErrDatabaseError                = errors.New("db error while querying for resource")
+	ErrResourceAlreadyExists        = errors.New("resource already exists")
+	ErrInvalidInventoryResourceID   = errors.New("inventory ID does not exist")
+	ErrInvalidInventoryResourceType = errors.New("invalid resource type")
 )
 
 type Usecase struct {
-	repository         ResourceRepository
-	Authz              authzapi.Authorizer
-	Eventer            eventingapi.Manager
-	Namespace          string
-	log                *log.Helper
-	Server             server.Server
-	DisablePersistence bool
+	repository                  ResourceRepository
+	inventoryResourceRepository InventoryResourceRepository
+	Authz                       authzapi.Authorizer
+	Eventer                     eventingapi.Manager
+	Namespace                   string
+	log                         *log.Helper
+	Server                      server.Server
+	DisablePersistence          bool
 }
 
-func New(repository ResourceRepository, authz authzapi.Authorizer, eventer eventingapi.Manager, namespace string, logger log.Logger, disablePersistence bool) *Usecase {
+func New(repository ResourceRepository, inventoryResourceRepository InventoryResourceRepository, authz authzapi.Authorizer, eventer eventingapi.Manager, namespace string, logger log.Logger, disablePersistence bool) *Usecase {
 	return &Usecase{
-		repository:         repository,
-		Authz:              authz,
-		Eventer:            eventer,
-		Namespace:          namespace,
-		log:                log.NewHelper(logger),
-		DisablePersistence: disablePersistence,
+		repository:                  repository,
```

Comment:
What do you think of renaming this to ReporterResourceRespository?

Reply:
SGTM - I'll get that updated

Reply:
updated!